packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart

```diff
@@ -254,7 +290,7 @@ class FlutterWebEntrypointBuilder implements Builder {
   Future<void> build(BuildStep buildStep) async {
     bool matches = false;
     for (String target in targets) {
-      if (buildStep.inputId.path.contains(target)) {
+      if (buildStep.inputId.path.contains(fs.path.setExtension(target, '_web_entrypoint.dart'))) {
```

Comment:
Would this not result in `blah._web_entrypoint.dart`?

Reply:
Surprisingly not